File: backend/app/models/tournament.py
from pydantic import BaseModel
from typing import List

class TournamentCreate(BaseModel):
    sport: str
    venue: str
    date: str
    number_of_teams: int

class TournamentResponse(TournamentCreate):
    id: str
    organizer_id: str
    teams_ids: List[str]

File: frontend/lib/models/tournament.dart
class Tournament {
  final String id;
  final String sport;
  final String venue;
  final String date;
  final int numberOfTeams;
  final String organizerId;

  Tournament({
    required this.id, required this.sport, required this.venue,
    required this.date, required this.numberOfTeams, required this.organizerId
  });

  factory Tournament.fromJson(Map<String, dynamic> json) => Tournament(
    id: json['id'], sport: json['sport'], venue: json['venue'],
    date: json['date'], numberOfTeams: json['number_of_teams'], organizerId: json['organizer_id']
  );
}
